lasagne/layers/input.py

```diff
@@ -67,6 +67,9 @@ def __init__(self, shape, input_var=None, name=None, **kwargs):
             if input_var.ndim != ndim:
                 raise ValueError("shape has %d dimensions, but variable has "
                                  "%d" % (ndim, input_var.ndim))
+            broad_pattern = [s == 1 for s in self.shape]
+            input_var = T.patternbroadcast(input_var, broad_pattern)
```

Comment:
> The code indeed would break after this PR, but could we have an extra parameter for the `InputLayer`, which by default is False and would broadcast the Variable if only set to True by the user?

Mhm, but then what's the advantage over adding the broadcast in `get_output()`, when collecting the input variables? We cannot catch all cases in the `InputLayer` constructor anyway. If a user supplies the input variables to `lasagne.layers.get_output()` instead, which is another common use pattern, then the `InputLayer` cannot influence them in any way.